mackerel-plugin-aws-lambda/lib/aws-lambda.go

```diff
@@ -0,0 +1,248 @@
+	}
+
+	datapoints := response.Datapoints
+	if len(datapoints) == 0 {
+		return nil, nil
+	}
+
+	latest := new(time.Time)
+	var latestDp *cloudwatch.Datapoint
+	for _, dp := range datapoints {
+		if dp.Timestamp.Before(*latest) {
+			continue
+		}
+
+		latest = dp.Timestamp
+		latestDp = dp
+	}
+
+	return latestDp, nil
+}
+
+// TransformMetrics converts some of datapoints to post differences of two metrics
+func transformMetrics(stats map[string]interface{}) map[string]interface{} {
+	// Although stats are interface{}, those values from cloudwatch.Datapoint are guaranteed to be float64.
+	if totalCount, ok := stats["invocations_total"].(float64); ok {
+		if errorCount, ok := stats["invocations_error"].(float64); ok {
+			stats["invocations_success"] = totalCount - errorCount
+		} else {
+			stats["invocations_success"] = totalCount
+		}
+		delete(stats, "invocations_total")
+	}
+	return stats
+}
+
+func mergeStatsFromDatapoint(stats map[string]interface{}, dp *cloudwatch.Datapoint, mg metricsGroup) map[string]interface{} {
+	if dp != nil {
+		for _, met := range mg.Metrics {
+			switch met.Type {
+			case metricsTypeAverage:
+				stats[met.MackerelName] = *dp.Average
+			case metricsTypeSum:
+				stats[met.MackerelName] = *dp.Sum
+			case metricsTypeMaximum:
+				stats[met.MackerelName] = *dp.Maximum
+			case metricsTypeMinimum:
+				stats[met.MackerelName] = *dp.Minimum
+			}
+		}
+	}
+	return stats
+}
+
+var lambdaMetricsGroup = []metricsGroup{
+	{CloudWatchName: "Invocations", Metrics: []metric{
+		{MackerelName: "invocations_total", Type: metricsTypeSum},
+	}},
+	{CloudWatchName: "Errors", Metrics: []metric{
+		{MackerelName: "invocations_error", Type: metricsTypeSum},
+	}},
+	{CloudWatchName: "DeadLetterErrors", Metrics: []metric{
+		{MackerelName: "dead_letter_errors", Type: metricsTypeSum},
+	}},
+	{CloudWatchName: "Throttles", Metrics: []metric{
+		{MackerelName: "invocations_throttles", Type: metricsTypeSum},
+	}},
+	{CloudWatchName: "Duration", Metrics: []metric{
+		{MackerelName: "duration_avg", Type: metricsTypeAverage},
+		{MackerelName: "duration_max", Type: metricsTypeMaximum},
+		{MackerelName: "duration_min", Type: metricsTypeMinimum},
+	}},
+}
+
+// FetchMetrics fetch the metrics
+func (p LambdaPlugin) FetchMetrics() (map[string]interface{}, error) {
+	stats := make(map[string]interface{})
+
+	for _, met := range lambdaMetricsGroup {
+		v, err := getLastPointFromCloudWatch(p.CloudWatch, p.FunctionName, met)
+		if err == nil {
+			stats = mergeStatsFromDatapoint(stats, v, met)
+		} else {
+			log.Printf("%s: %s", met, err)
+		}
+	}
+	return transformMetrics(stats), nil
+}
+
+// GraphDefinition of LambdaPlugin
+func (p LambdaPlugin) GraphDefinition() map[string]mp.Graphs {
+	labelPrefix := strings.Title(p.Prefix)
+
+	graphdef := map[string]mp.Graphs{
+		"invocations": {
+			Label: (labelPrefix + " Invocations"),
+			Unit:  "integer",
+			Metrics: []mp.Metrics{
+				{Name: "invocations_success", Label: "Success", Stacked: true},
```

Comment:
[may]In AWS integration of Mackerel, default is `Stacked: false`. I think it would be good if both have the same default value.